### PR TITLE
A batch of Doxygen updates

### PR DESCRIPTION
**Pull request by gheber**
Bot much to write home about, but a start.

### CODE REVIEW

File: src/hermes_types.h [hunk 1]
```diff
@@ -28,227 +28,277 @@
...
-typedef uint64_t u64;
-typedef int8_t i8;
-typedef int16_t i16;
-typedef int32_t i32;
-typedef int64_t i64;
-typedef float f32;
-typedef double f64;
-
-typedef u16 DeviceID;
+  typedef uint8_t u8;
```
Comment (jya-kmu) [inline src/hermes_types.h:31]: No indentation after `namespace` by [styleguide](https://google.github.io/styleguide/cppguide.html#Namespaces)

File: src/hermes_types.h [hunk 2]
```diff
@@ -28,227 +28,277 @@
...
-constexpr int kMaxPathLength = 256;
-constexpr int kMaxBufferPoolShmemNameLength = 64;
-constexpr int kMaxDevices = 8;
-constexpr int kMaxBucketNameSize = 256;
-constexpr int kMaxVBucketNameSize = 256;
-
-constexpr char kPlaceInHierarchy[] = "PlaceInHierarchy";
-
-#define HERMES_NOT_IMPLEMENTED_YET \
+  typedef std::vector<unsigned char> Blob;
```
Comment (jya-kmu) [inline src/hermes_types.h:45]: No indentation after `namespace` by [styleguide](https://google.github.io/styleguide/cppguide.html#Namespaces)

File: doc/Doxyfile.in [hunk 3]
```diff
@@ -1,12 +1,11 @@
 # Doxyfile 1.8.14
-
 #---------------------------------------------------------------------------
 # Project related configuration options
 #---------------------------------------------------------------------------
 DOXYFILE_ENCODING      = UTF-8
 PROJECT_NAME           = Hermes
-PROJECT_NUMBER         =
-PROJECT_BRIEF          =
+PROJECT_NUMBER         = 1.0.0-beta
```
Comment (ChristopherHogan) [inline doc/Doxyfile.in:7]: I thought we decided on `0.1.0-beta`.

**Review by ChristopherHogan — APPROVED** [review]

**Review by jya-kmu — APPROVED** [review]

File: doc/Doxyfile.in (hunk 3 above)
Comment (gheber) [reply doc/Doxyfile.in:7]: Whoops. Just saw this. Will fix.

**Pull request merged**